components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -478,6 +489,35 @@ && isStepHasMultiOption(context)
         }
     }
 
+    protected boolean isApplicationEnabled(HttpServletRequest request, AuthenticationContext context)
+            throws FrameworkException {
+
+        String type = request.getParameter(TYPE);
+        String clientId = request.getParameter(FrameworkConstants.RequestParams.ISSUER);
+        if (StringUtils.isBlank(type)) {
+            type = context.getRequestType();
+        }
+        if (StringUtils.isBlank(clientId)) {
+            clientId = request.getParameter(CLIENT_ID);
+        }
```

Comment:
lets remove this check. this make framework protocol specific as client_id is only available for oauth based apps. since in other places we can resolve without resorting to client id in here also the same should suffice.